src/Netscript/NetscriptHelpers.tsx

```diff
@@ -223,7 +223,7 @@ function spawnOptions(ctx: NetscriptContext, threadOrOption: unknown): CompleteS
   if (spawnDelay !== undefined) {
     result.spawnDelay = number(ctx, "spawnDelay", spawnDelay);
     if (result.spawnDelay < 0) {
-      throw errorMessage(ctx, `spawnDelay must be non-negative, got ${spawnDelay}`);
+      throw errorMessage(ctx, `spawnDelay must be non-negative, got ${String(spawnDelay)}`);
```

Comment:
After looking at a bunch of these... I'm not actually certain this is a good lint rule for our codebase.

Specifically, what this is triggering on a lot is `error  Invalid type "unknown" of template literal expression`. But this is precisely the time when we frequently *want* to use template literal expressions, particularly in error messages. I don't think wrapping these in `String()` adds anything, rather I think it obfuscates to little purpose.

If we can add an exception for "unknown" that would be good, but it might be too core of a type to do that. (And while there are flags for "allowAny" there isn't one for "allowUnknown". In that case, we might be better off without this rule entirely.

Reply:
Isn't this incorrect typing? It has nothing to do with the stringification. This is the entire context:

```ts
const { spawnDelay } = threadOrOption as Unknownify<CompleteSpawnOptions>;
  if (spawnDelay !== undefined) {
    result.spawnDelay = number(ctx, "spawnDelay", spawnDelay);
    if (result.spawnDelay < 0) {
      throw errorMessage(ctx, `spawnDelay must be non-negative, got ${spawnDelay}`);
    }
  }
```

Unknownify is being used plenty in Netscript Helpers to assert a certain sub-shape from the user-space where it is 'unknown' by default, then it is narrowed to a "type", and usually that type has optional params. So when it is implied by structure that a parameter is likely present, you can re-type it, guard it or do this unknownify trick.

In this case the type guard for undefined narrows it incorrectly to the **type** {} | null (from {} | undefined | null via unknown) and then explicitly validated and cast with another guard `number` which returns a new variable (but doesn't guard the original).

So the correct usage would be

```suggestion
      throw errorMessage(ctx, `spawnDelay must be non-negative, got ${result.spawnDelay}`);
```